### PR TITLE
Fix up issues in the PS Performance script

### PR DESCRIPTION
**Pull request by ThadHouse**
Timeout added to local run (configurable)
Local matcher now prints error on failure
Release mode as default
Automatically configure tls to match system if not set
Validate sets for tls and arch to enable tab completion

### CODE REVIEW

File: scripts/performance-helper.psm1 [hunk 1]
```diff
@@ -360,7 +361,14 @@ function Invoke-LocalExe {
     $FullCommand = "$Exe $RunArgs"
     Write-Debug "Running Locally: $FullCommand"
 
-    return (Invoke-Expression $FullCommand) -join "`n"
+    $LocalJob = Invoke-Command -ScriptBlock { & $Exe ($RunArgs).Split(" ") } -AsJob
+
+    # Wait 60 seconds for the job to finish
+    Wait-Job -Job $LocalJob -Timeout 60 | Out-Null
```
Comment (nibanks) [inline scripts/performance-helper.psm1:367]: Timeout should be an (optional) input argument, all the way up to the user when it calls the script.

Comment (ThadHouse) [reply scripts/performance-helper.psm1:367]: The timeout would be per individual test, not the whole script. But that'd be fine.

File: scripts/performance.ps1 [hunk 2]
```diff
@@ -102,6 +111,22 @@ if ($IsWindows) {
...
+    } else {
+        $LocalTls = "openssl"
+        $RemoteTls = $LocalTls
+    }
+} elseif (($LocalTls -ne "") -xor ($RemoteTls -ne "")) {
+    Write-Error "Both TLS arguments must be set if a manual setting is done"
+}
+
+Write-Host $LocalTls
+Write-Host $RemoteTls
```
Comment (nibanks) [inline scripts/performance.ps1:128]: Do we really need these?

Comment (ThadHouse) [reply scripts/performance.ps1]: Nope. Removed

**Pull request merged**